### PR TITLE
Extra settings for limiting search facets 10570

### LINKED ISSUE #10570: More indexing and searching improvements
This is a quick placeholder issue for making a PR with some incremental improvements I was able to come up with as part of an overall performance investigation in #10469 that are not covered in the ongoing PRs by others.

### PR DESCRIPTION
**Pull request by landreev**
**What this PR does / why we need it**:

This gives an instance admin some extra flexibility in limiting the search facets, as shown on the pages, to users and/or bots. Could be another useful tool to have in case of a search engine disaster. Explained in the guide. 

**Which issue(s) this PR closes**:

Closes #10570

**Special notes for your reviewer**:

**Suggestions on how to test this**:

**Does this PR introduce a user interface change? If mockups are available, please link/include them here**:

**Is there a release notes update needed for this change?**:

**Additional documentation**:

### CODE REVIEW

File: doc/sphinx-guides/source/installation/config.rst [hunk 1]
```diff
@@ -3860,6 +3860,23 @@ To enable the setting::
   curl -X PUT -d true "http://localhost:8080/api/admin/settings/:DisableSolrFacets"
 
 
+:DisableSolrFacetsForGuestUsers
++++++++++++++++++++++++++++++++
+
+Similar to the above, but will disable the facets for Guest users only.
```
Comment (pdurbin) [inline doc/sphinx-guides/source/installation/config.rst:3866]:
```suggestion
Similar to the above, but will disable the facets for Guest (unauthenticated) users only. 
```

File: doc/sphinx-guides/source/installation/config.rst [hunk 2]
```diff
@@ -3860,6 +3860,23 @@ To enable the setting::
   curl -X PUT -d true "http://localhost:8080/api/admin/settings/:DisableSolrFacets"
 
 
+:DisableSolrFacetsForGuestUsers
++++++++++++++++++++++++++++++++
+
+Similar to the above, but will disable the facets for Guest users only. 
+
+:DisableSolrFacetsForAnonymousUsers
```
Comment (pdurbin) [inline doc/sphinx-guides/source/installation/config.rst:3868]: What about `:DisableSolrFacetsForInitialRequests` instead?

Perhaps I'm misunderstanding the paragraph below, but when set to true, people would be affected by this (and not see the facets) the first time they visit a Dataverse installation, right? Unfortunately, these people would get a less rich, facetless experience, if I understand correctly, when this is set to true for the purpose of mostly avoiding loads from bots.

**Review by pdurbin — COMMENTED** [review]
Overall, this seems great. I left a comment about naming and docs. I could simply be confused.

File: doc/sphinx-guides/source/installation/config.rst (hunk 2 above)
Comment (pdurbin) [reply doc/sphinx-guides/source/installation/config.rst:3868]: Sure, `:DisableSolrFacetsWithoutJsession` (or whatever cApiTaLization) sounds good to me. 😄

**Comment by qqmyers** [comment]
@landreev - a couple things to get this through review:
- I see @pdurbin had one requested name change for a flag that needs to be done (looks like you are OK with the suggestion)
- the guides build is reporting a bad hyperlink
- looks like the IT tests didn't run last time - hopefully resolved with a merge with develop and/or your commits for other fixes.

**Review by qqmyers — APPROVED** [review]
Looks good. Build fails due to solr changes, so should be rerun as part of QA.

**Comment by landreev** [comment]
I was about to repeat that if there was any doubt whatsoever about this pr, it was entirely ok to leave it out of 6.3... but happy to see that it's been merged!

**Pull request merged**